src/booking/containers/booking-home/booking-home.component.ts

```diff
@@ -30,33 +31,54 @@ export class BookingHomeComponent implements OnInit, OnDestroy {
       bookingType: new FormControl(null)
     });
 
-    this.existingBookingsSubscription = this.bookingService.getBookings().subscribe(response => {
-      this.existingBookings = response.bookings;
-      this.assignBookingLocation();
+    this.existingBookingsSubscription = this.bookingService.getBookings().subscribe((arr) => {
+      this.locations$ = arr.bookings.map(
+        booking =>  this.bookingService.getBookingLocation(booking.base_location_id).pipe(
+          map( location => {
+            return {
+              ...booking,
+              locationName: location[0] && !!location[0].building_location_name && location[0].building_location_name  } ;
```

Comment:
Why you have the double bang check as well as the location name?

Reply:
checking for undefined, if it's not undefined that return from service then assign location name

Reply:
Does that mean there instances where it would be undefined (as opposed to null)?

Reply:
no Now we are using mock service but actual service implementation may return null or undefined. So good to check undefined as a standard way as in >=ES5  .

Reply:
Double bang removed as per discussion so resolving conversation